### DISCUSSION TITLE
RaspberryPI MAX485 adapter needs control pin for Serial

### DISCUSSION THREAD

**Discussion by EvilBeaver** [pymodbus-dev/pymodbus#1948] [Help] [answered]
Hello. I want to use RaspberryPI as Modbus master and I'm using MAX485 RS485 to TTL converter for connecting to bus.
This converter board connects to serial port and uses GPIO control pin for switching between send and receive serial data. Pin must be set to 1 when sending and to 0 when receiving data via serial port.

Question: does Pymodbus library has any options to manipulate GPIO pins when sending and receiving data? I need to perform pin on/off during modbus communications and if library does sending AND receiving in a single python API method then I need somehow to interfere into this process. Any advices?

**Comment by janiversen** [comment]
this is not something pymodbus deal with, we use pyserial, and I have no idea if that can do what you want.

However your setup is quite unusual, I have a rpi which I use for different tests and a serial rs485 adapter as well as a usb rs485, both work without any problems.

You can use send the request and then poll with the library, the mixin methods are just for convinience......anyhow if you need to change code it should be in pyserial.

**Reply by EvilBeaver** [reply]
Thanks for the answer. USB adapters work well, I know. Question is about non-USB. Probably I should switch to a USB dongle, or patch a serial library somehow

**Reply by janiversen** [reply, marked as answer]
your choice but not really a pymodbus problem.

I wrote that I have a serial adapter and a usb adapter which both works, the serial adapter, when use, is wired to a couple of pins. So I am aware that your problem is not a usb problem.

**Reply by EvilBeaver** [reply]
Can you tell which serial adapter do you use? What wiring does it have? Thanks in advance

**Reply by janiversen** [reply]
It's a no-name chinese one (from ali-express), and it is wired as any serial adapter.

But if you buy a new one, then buy a usb, it's more stable on the rs485 side (due to having a power connection through usb) and mostly with a faster controller.
